lib/src/rules/unreachable_from_main.dart

```diff
@@ -271,9 +271,13 @@ class _ReferenceVisitor extends RecursiveAstVisitor {
       return;
     }
 
-    if (node.type?.alias != null) {
-      // Any reference to a typedef counts as "reachable", since structural
-      // typing is used to match against objects.
+    // Any reference to a typedef is reachable, since structural typing is used
+    // to match against objects.
+    //
+    // The return type of an external variable declaration is reachable, since
+    // the external implementation can instantiate it.
+    if (node.type?.alias != null ||
+        node.isInExternalVariableTypeOrFunctionReturnType) {
```

Comment:
Do we care to also check whether the external definition is also reachable?

Reply:
The visitor should only be checking the external definition if it was reachable from main.